### PR TITLE
Handle entity resolver errors by logging and raising.

### LINKED ISSUE #289: Stop transform on Entity Resolver error
It currently just logs the error.

The error should be logged and the transform should be interrupted. When running with composite ETL, the SPARQL file should not be written and load should be skipped.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #289

### CODE REVIEW

File: lib/rialto/etl/service_client/entity_resolver.rb [hunk 1]
```diff
@@ -39,7 +40,8 @@ def resolve(type, params)
             raise "Entity resolver returned #{resp.status} for #{type} type and #{params} params."
           end
         rescue StandardError => exception
-          warn "Error resolving with path #{path}: #{exception.message}"
+          logger.warn "Error resolving with path #{path}: #{exception.message}"
+          raise
```
Comment (jcoyne) [inline lib/rialto/etl/service_client/entity_resolver.rb]: should we be re-raising the original exception here?

Comment (jcoyne) [reply lib/rialto/etl/service_client/entity_resolver.rb]: This does appear to reraise the original 👍

File: spec/service_client/entity_resolver_spec.rb [hunk 2]
```diff
@@ -32,16 +32,14 @@
...
-    # rubocop:disable Metrics/LineLength
     context 'when resolver behaves unexpectedly' do
       before do
         stub_request(:get, 'http://127.0.0.1:3001/person?full_name=Wilson,%20Jennifer%20L.&orcid_id=0000-0002-2328-2018')
           .to_return(status: 500, body: '')
       end
-      it 'prints a warning' do
-        expect { resolve }.to output('Error resolving with path person?orcid_id=0000-0002-2328-2018&full_name=Wilson%2C+Jennifer+L.: Entity resolver returned 500 for person type and {"orcid_id"=>"0000-0002-2328-2018", "full_name"=>"Wilson, Jennifer L."} params.' + "\n").to_stderr
+      it 'raises error' do
+        expect { resolve }.to raise_error(RuntimeError)
```
Comment (jcoyne) [inline spec/service_client/entity_resolver_spec.rb]: add a test that `logger` receives `warn`?

**Comment by jcoyne** [comment]
@justinlittman needs a rebase.

**Comment by justinlittman** [comment]
@jcoyne Rebased.

**Comment by mjgiarlo** [comment]
I have resolved the CodeClimate issues, and this PR should now/soon be :green_heart:

File: spec/service_client/entity_resolver_spec.rb (hunk 2 above)
Comment (justinlittman) [reply spec/service_client/entity_resolver_spec.rb]: @jcoyne Test added.

**Pull request merged**